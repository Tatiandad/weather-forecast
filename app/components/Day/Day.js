import React, { Component } from 'react';
import Moment from 'moment';
import Collapsible from 'react-collapsible';
import ClassNames from 'classnames';

const icon = require('../../images/white-arrow-transparent-png-22.png');
const windIcon = require('../../images/compass.png');

class Day extends Component {
  state = {
    open: false
  }

  degToCompass = (num) => {
    var val = Math.floor((num / 22.5) + 0.5);
    var arr = ["N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE", "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"];
    return arr[(val % 16)];
  }

  toggleHeader = () => {
    this.setState({ open: !this.state.open });
  }

  hourlyForecast = () => {
    const windUnit = this.props.unit === 'F' ? 'mph' : 'm/sec';
    const rows = this.props.data.map((set) => {
      const { wind, dt, dt_txt, weather, main  } = set;
      const windDir = this.degToCompass(wind.deg);
      const windIconDir = {transform: `rotate(${Math.round(wind.deg)}deg)`};
      return (
        <div key={`day-${dt}`} className='forecast-hourly-container'>
          <p>
            {Moment(dt_txt).format('HH:mm')}
          </p>
          <p>
            <img 
              className='forecast-hourly-icon' 
              src={`https://openweathermap.org/img/w/${weather[0].icon}.png`} 
              alt={weather[0].description}
            />
          </p>
          <p className='forecast-hourly-desc'>
            {weather[0].description}
          </p>
          <p>
            {Math.round(main.temp_min)}&deg;{this.props.unit} / {Math.round(main.temp_max)}&deg;{this.props.unit}
          </p>
          <p>
            <img className='forecast-hourly-wind-icon' src={windIcon} alt={windDir} style={windIconDir}/>
            {wind.speed}{windUnit}
          </p>
        </div>
      )
    });
    return (
      <div>
        {rows}
      </div>
    )
  }

  renderDayHeader = () => {
    const currentWeather = this.props.data[0];
    let date = Moment(this.props.date).format("dddd Do MMMM");
    if ( Moment(new Date()).format("dddd Do MMMM") === date) {
      date = 'Today'
    }
    
    return (
      <div className='forecast-hourly-header'>
        <p>
          <span className='forecast-hourly-date'>
            {date}
          </span>
          <img 
            className={ClassNames('forecast-hourly-icon', { 'hide': this.state.open })}
            src={`https://openweathermap.org/img/w/${currentWeather.weather[0].icon}.png`} 
            alt={currentWeather.weather[0].description}
          />
          <span className={ClassNames('forecast-hourly-temp', { 'hide': this.state.open })}>
            {Math.round(currentWeather.main.temp)}&deg;{this.props.unit}
          </span>
        </p>
        <img className='icon' src={icon} alt='trigger-icon'/>
      </div>
    )
  }

  render () {
    return (
      <div className='forecast-day'>
        <Collapsible 
          trigger={this.renderDayHeader()}
          triggerTagName='div'
          triggerStyle={{ backgroundColor: '#1b2d3a', color: '#ffffff', cursor: 'pointer' }}
          onOpen={this.toggleHeader}
          onClose={this.toggleHeader}
        >
          {this.hourlyForecast()}
        </Collapsible>
      </div>
    )
  }
}

export default Day;