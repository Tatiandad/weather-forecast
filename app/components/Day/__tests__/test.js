import React, { Component } from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import "babel-polyfill";
import { shallow, mount } from 'enzyme';
import Day from '../Day';
import forecast from '../../../utils/dummy-forecast';

configure({ adapter: new Adapter() });

let shallowWrapper;
let mountWrapper;

const dummyProps = {
  data: [forecast.list[0]],
  date: '2018-12-17',
  unit: 'F'
};

beforeEach(() => {
  shallowWrapper = shallow(<Day {...dummyProps}/>);
  mountWrapper = mount(<Day {...dummyProps}/>);
});

describe('<Day />', () => {
  it('renders <Day /> component', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('renders elements of <Day /> correctly', () => {
    const wrapperDiv = shallowWrapper.find('.forecast-day');
    const collapsible = shallowWrapper.find('Collapsible');
    const weatherIcon = shallowWrapper.find('.forecast-hourly-icon');
    const weatherDesc = shallowWrapper.find('.forecast-hourly-desc').at(0);
    expect(wrapperDiv.length).toBe(1);
    expect(collapsible.length).toBe(1);
    expect(weatherIcon.prop('alt')).toEqual(dummyProps.data[0].weather[0].description);
    expect(weatherDesc.text()).toEqual(dummyProps.data[0].weather[0].description);
  });

  it('toggles collapsible', () => {
    let collapsible = mountWrapper.find('.Collapsible__trigger').at(0);
    expect(collapsible.prop('className')).toEqual('Collapsible__trigger is-closed');
    collapsible.simulate('click');
    mountWrapper.update();
    collapsible = mountWrapper.find('.Collapsible__trigger').at(0);
    expect(collapsible.prop('className')).toEqual('Collapsible__trigger is-open');
  });
});