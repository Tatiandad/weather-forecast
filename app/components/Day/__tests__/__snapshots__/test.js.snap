// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> renders <Day /> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Day
    data={
      Array [
        Object {
          "clouds": Object {
            "all": 92,
          },
          "dt": 1544896800,
          "dt_txt": "2018-12-15 18:00:00",
          "main": Object {
            "grnd_level": 1007.09,
            "humidity": 98,
            "pressure": 1007.09,
            "sea_level": 1014.72,
            "temp": 5.15,
            "temp_kf": 0.01,
            "temp_max": 5.15,
            "temp_min": 5.13,
          },
          "rain": Object {
            "3h": 6.1725,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "moderate rain",
              "icon": "10n",
              "id": 501,
              "main": "Rain",
            },
          ],
          "wind": Object {
            "deg": 152.001,
            "speed": 9.02,
          },
        },
      ]
    }
    date="2018-12-17"
    unit="F"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Collapsible
        className=""
        classParentString="Collapsible"
        contentInnerClassName=""
        contentOuterClassName=""
        easing="linear"
        lazyRender={false}
        onClose={[Function]}
        onClosing={[Function]}
        onOpen={[Function]}
        onOpening={[Function]}
        open={false}
        openedClassName=""
        overflowWhenOpen="hidden"
        tabIndex={null}
        transitionCloseTime={null}
        transitionTime={400}
        trigger={
          <div
            className="forecast-hourly-header"
          >
            <p>
              <span
                className="forecast-hourly-date"
              >
                Today
              </span>
              <img
                alt="moderate rain"
                className="forecast-hourly-icon"
                src="http://openweathermap.org/img/w/10n.png"
              />
              <span
                className="forecast-hourly-temp"
              >
                5
                °
                F
              </span>
            </p>
            <img
              alt="trigger-icon"
              className="icon"
              src="white-arrow-transparent-png-22.png"
            />
          </div>
        }
        triggerClassName=""
        triggerDisabled={false}
        triggerOpenedClassName=""
        triggerSibling={null}
        triggerStyle={
          Object {
            "backgroundColor": "#1b2d3a",
            "color": "#ffffff",
            "cursor": "pointer",
          }
        }
        triggerTagName="div"
      >
        <div>
          <div
            className="forecast-hourly-container"
          >
            <p>
              18:00
            </p>
            <p>
              <img
                alt="moderate rain"
                className="forecast-hourly-icon"
                src="http://openweathermap.org/img/w/10n.png"
              />
            </p>
            <p
              className="forecast-hourly-desc"
            >
              moderate rain
            </p>
            <p>
              5
              °
              F
               / 
              5
              °
              F
            </p>
            <p>
              <img
                alt="SSE"
                className="forecast-hourly-wind-icon"
                src="compass.png"
                style={
                  Object {
                    "transform": "rotate(152deg)",
                  }
                }
              />
              9.02
              mph
            </p>
          </div>
        </div>
      </Collapsible>,
      "className": "forecast-day",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div
            className="forecast-hourly-container"
          >
            <p>
              18:00
            </p>
            <p>
              <img
                alt="moderate rain"
                className="forecast-hourly-icon"
                src="http://openweathermap.org/img/w/10n.png"
              />
            </p>
            <p
              className="forecast-hourly-desc"
            >
              moderate rain
            </p>
            <p>
              5
              °
              F
               / 
              5
              °
              F
            </p>
            <p>
              <img
                alt="SSE"
                className="forecast-hourly-wind-icon"
                src="compass.png"
                style={
                  Object {
                    "transform": "rotate(152deg)",
                  }
                }
              />
              9.02
              mph
            </p>
          </div>
        </div>,
        "className": "",
        "classParentString": "Collapsible",
        "contentInnerClassName": "",
        "contentOuterClassName": "",
        "easing": "linear",
        "lazyRender": false,
        "onClose": [Function],
        "onClosing": [Function],
        "onOpen": [Function],
        "onOpening": [Function],
        "open": false,
        "openedClassName": "",
        "overflowWhenOpen": "hidden",
        "tabIndex": null,
        "transitionCloseTime": null,
        "transitionTime": 400,
        "trigger": <div
          className="forecast-hourly-header"
        >
          <p>
            <span
              className="forecast-hourly-date"
            >
              Today
            </span>
            <img
              alt="moderate rain"
              className="forecast-hourly-icon"
              src="http://openweathermap.org/img/w/10n.png"
            />
            <span
              className="forecast-hourly-temp"
            >
              5
              °
              F
            </span>
          </p>
          <img
            alt="trigger-icon"
            className="icon"
            src="white-arrow-transparent-png-22.png"
          />
        </div>,
        "triggerClassName": "",
        "triggerDisabled": false,
        "triggerOpenedClassName": "",
        "triggerSibling": null,
        "triggerStyle": Object {
          "backgroundColor": "#1b2d3a",
          "color": "#ffffff",
          "cursor": "pointer",
        },
        "triggerTagName": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="forecast-hourly-container"
            >
              <p>
                18:00
              </p>
              <p>
                <img
                  alt="moderate rain"
                  className="forecast-hourly-icon"
                  src="http://openweathermap.org/img/w/10n.png"
                />
              </p>
              <p
                className="forecast-hourly-desc"
              >
                moderate rain
              </p>
              <p>
                5
                °
                F
                 / 
                5
                °
                F
              </p>
              <p>
                <img
                  alt="SSE"
                  className="forecast-hourly-wind-icon"
                  src="compass.png"
                  style={
                    Object {
                      "transform": "rotate(152deg)",
                    }
                  }
                />
                9.02
                mph
              </p>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "day-1544896800",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  18:00
                </p>,
                <p>
                  <img
                    alt="moderate rain"
                    className="forecast-hourly-icon"
                    src="http://openweathermap.org/img/w/10n.png"
                  />
                </p>,
                <p
                  className="forecast-hourly-desc"
                >
                  moderate rain
                </p>,
                <p>
                  5
                  °
                  F
                   / 
                  5
                  °
                  F
                </p>,
                <p>
                  <img
                    alt="SSE"
                    className="forecast-hourly-wind-icon"
                    src="compass.png"
                    style={
                      Object {
                        "transform": "rotate(152deg)",
                      }
                    }
                  />
                  9.02
                  mph
                </p>,
              ],
              "className": "forecast-hourly-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "18:00",
                },
                "ref": null,
                "rendered": "18:00",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="moderate rain"
                    className="forecast-hourly-icon"
                    src="http://openweathermap.org/img/w/10n.png"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "moderate rain",
                    "className": "forecast-hourly-icon",
                    "src": "http://openweathermap.org/img/w/10n.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "moderate rain",
                  "className": "forecast-hourly-desc",
                },
                "ref": null,
                "rendered": "moderate rain",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    5,
                    "°",
                    "F",
                    " / ",
                    5,
                    "°",
                    "F",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  5,
                  "°",
                  "F",
                  " / ",
                  5,
                  "°",
                  "F",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="SSE"
                      className="forecast-hourly-wind-icon"
                      src="compass.png"
                      style={
                        Object {
                          "transform": "rotate(152deg)",
                        }
                      }
                    />,
                    9.02,
                    "mph",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "SSE",
                      "className": "forecast-hourly-wind-icon",
                      "src": "compass.png",
                      "style": Object {
                        "transform": "rotate(152deg)",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  9.02,
                  "mph",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Collapsible
          className=""
          classParentString="Collapsible"
          contentInnerClassName=""
          contentOuterClassName=""
          easing="linear"
          lazyRender={false}
          onClose={[Function]}
          onClosing={[Function]}
          onOpen={[Function]}
          onOpening={[Function]}
          open={false}
          openedClassName=""
          overflowWhenOpen="hidden"
          tabIndex={null}
          transitionCloseTime={null}
          transitionTime={400}
          trigger={
            <div
              className="forecast-hourly-header"
            >
              <p>
                <span
                  className="forecast-hourly-date"
                >
                  Today
                </span>
                <img
                  alt="moderate rain"
                  className="forecast-hourly-icon"
                  src="http://openweathermap.org/img/w/10n.png"
                />
                <span
                  className="forecast-hourly-temp"
                >
                  5
                  °
                  F
                </span>
              </p>
              <img
                alt="trigger-icon"
                className="icon"
                src="white-arrow-transparent-png-22.png"
              />
            </div>
          }
          triggerClassName=""
          triggerDisabled={false}
          triggerOpenedClassName=""
          triggerSibling={null}
          triggerStyle={
            Object {
              "backgroundColor": "#1b2d3a",
              "color": "#ffffff",
              "cursor": "pointer",
            }
          }
          triggerTagName="div"
        >
          <div>
            <div
              className="forecast-hourly-container"
            >
              <p>
                18:00
              </p>
              <p>
                <img
                  alt="moderate rain"
                  className="forecast-hourly-icon"
                  src="http://openweathermap.org/img/w/10n.png"
                />
              </p>
              <p
                className="forecast-hourly-desc"
              >
                moderate rain
              </p>
              <p>
                5
                °
                F
                 / 
                5
                °
                F
              </p>
              <p>
                <img
                  alt="SSE"
                  className="forecast-hourly-wind-icon"
                  src="compass.png"
                  style={
                    Object {
                      "transform": "rotate(152deg)",
                    }
                  }
                />
                9.02
                mph
              </p>
            </div>
          </div>
        </Collapsible>,
        "className": "forecast-day",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <div
              className="forecast-hourly-container"
            >
              <p>
                18:00
              </p>
              <p>
                <img
                  alt="moderate rain"
                  className="forecast-hourly-icon"
                  src="http://openweathermap.org/img/w/10n.png"
                />
              </p>
              <p
                className="forecast-hourly-desc"
              >
                moderate rain
              </p>
              <p>
                5
                °
                F
                 / 
                5
                °
                F
              </p>
              <p>
                <img
                  alt="SSE"
                  className="forecast-hourly-wind-icon"
                  src="compass.png"
                  style={
                    Object {
                      "transform": "rotate(152deg)",
                    }
                  }
                />
                9.02
                mph
              </p>
            </div>
          </div>,
          "className": "",
          "classParentString": "Collapsible",
          "contentInnerClassName": "",
          "contentOuterClassName": "",
          "easing": "linear",
          "lazyRender": false,
          "onClose": [Function],
          "onClosing": [Function],
          "onOpen": [Function],
          "onOpening": [Function],
          "open": false,
          "openedClassName": "",
          "overflowWhenOpen": "hidden",
          "tabIndex": null,
          "transitionCloseTime": null,
          "transitionTime": 400,
          "trigger": <div
            className="forecast-hourly-header"
          >
            <p>
              <span
                className="forecast-hourly-date"
              >
                Today
              </span>
              <img
                alt="moderate rain"
                className="forecast-hourly-icon"
                src="http://openweathermap.org/img/w/10n.png"
              />
              <span
                className="forecast-hourly-temp"
              >
                5
                °
                F
              </span>
            </p>
            <img
              alt="trigger-icon"
              className="icon"
              src="white-arrow-transparent-png-22.png"
            />
          </div>,
          "triggerClassName": "",
          "triggerDisabled": false,
          "triggerOpenedClassName": "",
          "triggerSibling": null,
          "triggerStyle": Object {
            "backgroundColor": "#1b2d3a",
            "color": "#ffffff",
            "cursor": "pointer",
          },
          "triggerTagName": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="forecast-hourly-container"
              >
                <p>
                  18:00
                </p>
                <p>
                  <img
                    alt="moderate rain"
                    className="forecast-hourly-icon"
                    src="http://openweathermap.org/img/w/10n.png"
                  />
                </p>
                <p
                  className="forecast-hourly-desc"
                >
                  moderate rain
                </p>
                <p>
                  5
                  °
                  F
                   / 
                  5
                  °
                  F
                </p>
                <p>
                  <img
                    alt="SSE"
                    className="forecast-hourly-wind-icon"
                    src="compass.png"
                    style={
                      Object {
                        "transform": "rotate(152deg)",
                      }
                    }
                  />
                  9.02
                  mph
                </p>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "day-1544896800",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    18:00
                  </p>,
                  <p>
                    <img
                      alt="moderate rain"
                      className="forecast-hourly-icon"
                      src="http://openweathermap.org/img/w/10n.png"
                    />
                  </p>,
                  <p
                    className="forecast-hourly-desc"
                  >
                    moderate rain
                  </p>,
                  <p>
                    5
                    °
                    F
                     / 
                    5
                    °
                    F
                  </p>,
                  <p>
                    <img
                      alt="SSE"
                      className="forecast-hourly-wind-icon"
                      src="compass.png"
                      style={
                        Object {
                          "transform": "rotate(152deg)",
                        }
                      }
                    />
                    9.02
                    mph
                  </p>,
                ],
                "className": "forecast-hourly-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "18:00",
                  },
                  "ref": null,
                  "rendered": "18:00",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="moderate rain"
                      className="forecast-hourly-icon"
                      src="http://openweathermap.org/img/w/10n.png"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "moderate rain",
                      "className": "forecast-hourly-icon",
                      "src": "http://openweathermap.org/img/w/10n.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "moderate rain",
                    "className": "forecast-hourly-desc",
                  },
                  "ref": null,
                  "rendered": "moderate rain",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      5,
                      "°",
                      "F",
                      " / ",
                      5,
                      "°",
                      "F",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    5,
                    "°",
                    "F",
                    " / ",
                    5,
                    "°",
                    "F",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="SSE"
                        className="forecast-hourly-wind-icon"
                        src="compass.png"
                        style={
                          Object {
                            "transform": "rotate(152deg)",
                          }
                        }
                      />,
                      9.02,
                      "mph",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "SSE",
                        "className": "forecast-hourly-wind-icon",
                        "src": "compass.png",
                        "style": Object {
                          "transform": "rotate(152deg)",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    9.02,
                    "mph",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
